test/Atmos/EDMF/edmf_kernels.jl

```diff
@@ -698,6 +656,87 @@ function flux_first_order!(
         Σfluxes(eq_tends(en_ρaθ_liq_q_tot_cv(), atmos, tend), atmos, args)
 end;
 
+function precompute(::EDMF, bl, args, ts, ::Flux{FirstOrder})
+    @unpack state, aux = args
+    env = environment_vars(state, aux, n_updrafts(bl.turbconv))
+    ρa_up = compute_ρa_up(bl, state, aux)
+    return (; env, ρa_up)
+end
+
+function precompute(::EDMF, bl, args, ts, ::Flux{SecondOrder})
+    @unpack state, aux, diffusive, t = args
+    N_up = n_updrafts(bl.turbconv)
+    env = environment_vars(state, aux, N_up)
+    ts_en = new_thermo_state_en(bl, bl.moisture, state, aux, ts)
+    ts_up = new_thermo_state_up(bl, bl.moisture, state, aux, ts)
+
+    E_dyn, Δ_dyn, E_trb = entr_detr(bl, state, aux, ts_up, ts_en, env)
+
+    l_mix, ∂b∂z_env, Pr_t = mixing_length(
+        bl,
+        bl.turbconv.mix_len,
+        args,
+        Δ_dyn,
+        E_trb,
+        ts,
+        ts_en,
+        env,
+    )
+    ρa_up = compute_ρa_up(bl, state, aux)
```

Comment:
`ρa_up` and `env` are precomputed for the first order fluxes. Is there a way to unpack their values here instead of computing them again?

Reply:
No, not easily. `flux_first_order!`, `flux_second_order!` and `source!` are separate calls, so they can't easily share a cache. We'd need for `flux_first_order!` (assuming that's the first one called) to return the cache and then pass it into `flux_second_order!` and `source!` (which would change the interface).